frontend/common/src/helpers/router.tsx

```diff
@@ -119,6 +121,16 @@ export const useRouter = (
         // may or may not be a promise, so attempt to resolve it. A non-promise value will simply resolve to itself.
         const route = await Promise.resolve(routeMaybePromise);
 
+        // Redirect authenticated user if no email exists yet
+        if (
+          welcomeRoute &&
+          loggedIn &&
+          (typeof loggedInEmail === undefined || loggedInEmail === null)
```

Comment:
Is there any reason you're using `typeof` here instead of just checking if `loggedInEmail === undefined`?
Also btw we have a helper function for this in common/src/helpers/util.ts

Reply:
Oh cool, thanks for the tip! This is actually a bug and I didn't mean to write it this way but `typeof` is fully intended because of weirdness with javascript. 

```tsx
loggedInEmail === undefined // throws an error "loggedInEmail is not defined" since it is being used as a value here

typeof loggedInEmail === 'undefined' // used as type so will not throw an error
```